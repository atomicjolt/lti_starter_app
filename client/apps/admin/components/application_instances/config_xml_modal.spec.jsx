import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ConfigXmlModal from './config_xml_modal';

describe('config xml modal', () => {
  let result;
  let instance;
  let closed;

  const isOpen = true;
  const closeModal = () => { closed = true; };
  const application = {
    id: 1,
    name: 'bfcoder'
  };
  const applicationInstance = {
    id: 1,
    lti_key: 'something-special',
    lti_secret: '12345',
    lti_config_xml: 'IMA XML',
  };

  // https://medium.com/@amanverma.dev/mocking-create-portal-to-utilize-react-test-renderer-in-writing-snapshot-uts-c49773c88acd
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element) => element);
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  beforeEach(() => {
    closed = false;
    result = TestRenderer.create(
      <ConfigXmlModal
        isOpen={isOpen}
        closeModal={closeModal}
        application={application}
        applicationInstance={applicationInstance}
      />);
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the closeModal function', () => {
    expect(closed).toBeFalsy();
    instance.findByType('button').props.onClick();
    expect(closed).toBeTruthy();
  });
});
