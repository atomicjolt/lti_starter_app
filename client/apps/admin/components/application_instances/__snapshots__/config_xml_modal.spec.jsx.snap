// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config xml modal matches the snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="c-modal c-modal--settings is-open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Application Instances Modal"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="c-modal__background"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="c-modal__title"
  >
    Config XML for 
    bfcoder
     Instance
  </h2>
  <div
    className="o-grid__item u-half"
  >
    <Input
      className="c-input"
      inputProps={
        Object {
          "id": "lti_config_xml_lti_key",
          "name": "lti_key",
          "readOnly": true,
          "type": "text",
          "value": "something-special",
        }
      }
      labelText="LTI Key"
    />
  </div>
  <div
    className="o-grid__item u-half"
  >
    <Input
      className="c-input"
      inputProps={
        Object {
          "id": "lti_config_xml_lti_secret",
          "name": "lti_secret",
          "readOnly": true,
          "type": "text",
          "value": "12345",
        }
      }
      labelText="LTI Secret"
    />
  </div>
  <div
    className="o-grid__item u-full"
  >
    <Textarea
      className="c-input"
      labelText="LTI Config XML"
      textareaProps={
        Object {
          "id": "application_instance_lti_config_xml",
          "name": "lti_config_xml",
          "onChange": [Function],
          "readOnly": true,
          "rows": 20,
          "value": "IMA XML",
        }
      }
    />
  </div>
  <button
    className="c-btn c-btn--gray--large u-m-right"
    onClick={[Function]}
    type="button"
  >
    Close
  </button>
</Modal>
`;
